test/e2e/test_cap_kill.py

```diff
@@ -0,0 +1,113 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+"""
+This test module contains tests that verify the Worker agent removes CAP_KILL from its inheritable
+capability set
+"""
+
+import os
+import re
+
+import boto3
+import botocore
+import pytest
+
+from deadline_test_fixtures import (
+    DeadlineClient,
+    EC2InstanceWorker,
+    Job,
+    TaskStatus,
+)
+from e2e.conftest import DeadlineResources
+
+
+@pytest.mark.skipif(
+    os.environ["OPERATING_SYSTEM"] == "windows",
+    reason="Linux specific test",
+)
+def test_cap_kill_not_inherited_by_running_jobs(
+    session_worker: EC2InstanceWorker,
+    deadline_client: DeadlineClient,
+    deadline_resources: DeadlineResources,
+) -> None:
+    """Tests that the worker agent drops CAP_KILL from its inheritable capability set and that
+    session actions are not able to signal processes belonging to different OS users"""
+
+    # GIVEN
+    # Send SSM command to write and run a bash script
+    # The script creates a detached sleep process and outputs that process' PID
+    # This sleep process will run as the ssm-user which is different from the job user
+    result = session_worker.send_command(
+        " ; ".join(
+            [
+                "echo '#!/bin/bash' > script.sh",
+                "echo 'set -euo pipefail' >> script.sh",
+                "echo 'nohup sleep 120 < /dev/null 2> /dev/null > /dev/null &' >> script.sh",
```

Comment:
Two questions:
1. Is 120s sufficient? I'm thinking about potential test timing. e.g. if it, for some reason, takes longer than expected for the worker to pick up and run the submitted job.
2. Should we have an ssm command at the end of the test that explicitly kills the script, just to make sure that we're not leaving anything behind? There's *probably* no harm, right now, in the test leaving a script running for a short time, but if this becomes a pattern that is replicated then the replicant may clash with this test in random unexpected ways...

Reply:
I've bumped the sleep to 240s which should be sufficient. I've also moved the background sleep job to a pytest fixture which handles cleanup of the process